superset/models/helpers.py

```diff
@@ -1577,57 +1669,43 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                         col=granularity,
                     )
                 )
-            time_filters: List[Any] = []
+            time_filters = []
 
             if is_timeseries:
-                if isinstance(dttm_col, dict):
-                    timestamp = self.get_timestamp_expression(
-                        dttm_col, time_grain, template_processor=template_processor
-                    )
-                else:
-                    timestamp = dttm_col.get_timestamp_expression(
-                        time_grain=time_grain, template_processor=template_processor
-                    )
+                timestamp = dttm_col.get_timestamp_expression(
```

Comment:
@hughhhh this logic causes an error if the `dttm_col` is associated with a query as opposed to a table as `dttm_col` which has been coerced to be of type `TableColumn` has no table associated with it, i.e., `dttm_col.table` is `None` and thus [this](https://github.com/apache/superset/blob/2222073778b0cee193f34c2500f2c489bb2a4bbe/superset/connectors/sqla/models.py#L265-L267) throws an exception.